client/src/game/star.js

```diff
@@ -589,24 +618,27 @@ class Star extends EventEmitter {
         // this.setScale()
       }
     }
+
+    //Redraw
+    this.draw()
   }
 
   updateVisibility() {
-    const displayTextZoom = 150
-
     this.graphics_star.visible = !this.hasSpecialist()
-    this.graphics_shape_part.visible = this.zoomPercent > displayTextZoom
-    this.graphics_shape_full.visible = this.zoomPercent <= displayTextZoom
-    this.graphics_shape_part_warp.visible = this.zoomPercent > displayTextZoom && this.data.warpGate
-    this.graphics_shape_full_warp.visible = this.zoomPercent <= displayTextZoom && this.data.warpGate
-    this.graphics_hyperspaceRange.visible = this.isSelected// && this.zoomPercent < 100
-    this.graphics_scanningRange.visible = this.isSelected// && this.zoomPercent < 100
-    this.graphics_natural_resources_ring.visible = this._isInScanningRange() && this.zoomPercent > displayTextZoom
-
-    if (this.text_name) this.text_name.visible = this.isSelected || this.zoomPercent > displayTextZoom
-    if (this.container_planets) this.container_planets.visible = this._isInScanningRange() && this.zoomPercent > displayTextZoom
-    if (this.text_infrastructure) this.text_infrastructure.visible = this.isMouseOver || this.isSelected || this.zoomPercent > displayTextZoom
-    if (this.text_garrison) this.text_garrison.visible = this.data.infrastructure && (this.isSelected || this.isMouseOver || this.zoomPercent > displayTextZoom)
+    this.graphics_hyperspaceRange.visible = this.isSelected
+    this.graphics_scanningRange.visible = this.isSelected
+    this.graphics_natural_resources_ring.visible = this._isInScanningRange() && this.zoomDepth >= 4
```

Comment:
You also need to consider **planets**, not just the solid ring. Planets should be drawn at the same zoom depth as the natural resources ring.